java-dynamic-sqs-listener-core/src/main/java/com/jashmore/sqs/broker/concurrent/properties/CachingConcurrentMessageBrokerProperties.java

```diff
@@ -0,0 +1,33 @@
+package com.jashmore.sqs.broker.concurrent.properties;
+
+import com.jashmore.sqs.util.Suppliers;
+
+import java.util.function.Supplier;
+import javax.validation.constraints.Min;
+
+/**
+ * Implementation that will cache the values as the methods to retrieve the values may be costly.
+ *
+ * <p>For example, an outbound call is needed to get this value and it is costly to do this every time a message has been processed.
```

Comment:
What outbound call? Isn't this just to a properties object? Does the value ever change? The memoize has no expiry correct?

Reply:
So I only provide an implementation this is static but you could implement the FeatureFlagConcurrentMessageBrokerProperties that calls out to LaunchDarkly. To make sure you don't make a call out to LaunchDarkly at the end of every message this Caching implementation will memoize the value for the provided milliseconds.

I can extend this documentation more to make this clear. You can see an example of it memoizing in the example where my implementation of the ConcurrentMessageBrokerProperties uses Random to randomly change the level of concurrency.